### ISSUE TITLE
[sym7] Make:user not compatible with phpstan

### ISSUE THREAD

**Issue by Xbirdfr** [symfony/maker-bundle#1404]
Hi !

when using the maker bundle to create user auth system, phpstan seem not very happy with what you get

To reproduce, do this : (I use the symfony cli on version 5.7.4)

`symfony new --webapp .`
`
symfony console make:user`

with all parameters by default

```
The name of the security user class (e.g. User) [User]:
 > 

 Do you want to store user data in the database (via Doctrine)? (yes/no) [yes]:
 > 

 Enter a property name that will be the unique "display" name for the user (e.g. email, username, uuid) [email]:
 > 

 Will this app need to hash/check user passwords? Choose No if passwords are not needed or will be checked/hashed by some other system (e.g. a single sign-on server).

 Does this app need to hash/check user passwords? (yes/no) [yes]:
 >
```
then add phpstan

`composer require --dev phpstan/phpstan`

and answer yes when the recipe ask execution

_recipe : https://github.com/symfony/recipes-contrib/tree/main/phpstan/phpstan/1.0_

this add the phpstan.dist.neon file, with "level: 6" check

finally, execute phpstan : 

`phpstan analyse ./src`

and you got errors :

```
Note: Using configuration file E:\_workspace\symfony7-test\phpstan.dist.neon.
 3/3 [============================] 100%

 ------ --------------------------------------------------------------------------------------
  Line   Entity\User.php

 ------ --------------------------------------------------------------------------------------
  :22    Property App\Entity\User::$roles type has no value type specified in iterable    

         type array.

         💡 See:

            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type
  :69    Method App\Entity\User::setRoles() has parameter $roles with no value type       

         specified in iterable type array.

         💡 See:

            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type
 ------ --------------------------------------------------------------------------------------

 ------ --------------------------------------------------------------------------------- 
  Line   Repository\UserRepository.php
 ------ --------------------------------------------------------------------------------- 
  :21    PHPDoc tag @implements contains generic type
         Symfony\Component\Security\Core\User\PasswordUpgraderInterface<App\Entity\User>  
         but interface Symfony\Component\Security\Core\User\PasswordUpgraderInterface is  
         not generic.
 ------ --------------------------------------------------------------------------------- 

                                                                                          
 [ERROR] Found 3 errors
```

**Comment by Xbirdfr** [comment]
I think I can fix `Entity\User.php` with : 

line 22 : 
```
/**
     * User's roles
     *
     * @var array<string>
     */
    #[ORM\Column]
    private array $roles = [];
```
and line 74 : 

```
**
     * Add user's role(s)
     *
     * @param array<string> $roles
     * @return static
     */
    public function setRoles(array $roles): static
    {
        $this->roles = $roles;

        return $this;
    }

```

but I don't know if this is a good way to fix this, also I don't know how to fix `Repository\UserRepository.php`

**Comment by Xbirdfr** [comment]
seem related to https://github.com/symfony/symfony/commit/d473b909e8a77a2bb03c94a4f6564f2e020f09e7

**Comment by Xbirdfr** [comment]
Seem fixed by this PR https://github.com/symfony/maker-bundle/pull/1402/files

**Comment by maelanleborgne** [comment]
Can be closed now that #1402 and #1411 have been merged
